### PR TITLE
smartmontools: build without c++11 on old systems

### PR DESCRIPTION
**Pull request by kencu**
closes: https://trac.macports.org/ticket/55730

### CODE REVIEW

File: sysutils/smartmontools/Portfile [hunk 1]
```diff
@@ -31,6 +31,13 @@ configure.args      --without-savestates \
                     --enable-sample \
                     --with-libcap-ng=no
 
+platform darwin {
+    if {${configure.cxx_stdlib} eq "libstdc++"} {
```
Comment (RJVB) [inline sysutils/smartmontools/Portfile]: Shouldn't this use `[string match "*libstdc\+\+" ${configure.cxx_stdlib}]`? When I checked earlier today if using libstdc++ was a viable workaround noticed that "base" now selects "macports-libstdc++" when you pass `configure.compiler=macports-gcc-N`.

If I understand the raison d'être for that approach (avoid confusion between the system libstdc++ and the MacPorts copy) it should be used on legacy, libstdc++-based systems too.

Comment (kencu) [reply sysutils/smartmontools/Portfile]: I copied and pasted that directly out of your recommendation in the other PR. You're killing me here, Rene.

**Comment by RJVB** [comment]
Apologies, you must have copied from an older post. I only discovered this morning that macports-libstdc++ is now an official thing and updated my review of you earlier PR today to use this [string match] expression.

**Comment by RJVB** [comment]
>But your string match would disable `c++11` if `cxx_stdlib` = `macports-libstdc++`, and that is not correct.

/methinks that's the point here... use the option to tell the buildsystem not even to try if the optional C++11 could be built. And that's exactly what's going to happen if your last claim is correct.

>`macports-libstdc++` specifically exists to support c++11.

Yes, newer libstdc++ versions than what Apple provide support C++11. I've been using that long before the macports-libstdc++ trick was designed.
Macports-libstdc++ exists so the libstdc++ provided by MacPorts is used explicitly, something a priori that should happen automatically when you use a GCC version from MacPorts.
>
>Anyway, without using the `cxx11 1.1 PG`, `cxx_stdlib` can never eq `macports-libstdc++` anyway, unless someone is playing around.

Well, I did not change anything in "base" w.r.t. this and yet it gave me exactly that when I passed only `configure.compiler=macports-gcc-7` on the commandline. I had to add `configure.cxx_lib=libstdc++` in addition to make it use libstdc++. Impossible to miss this because I never installed the macports-libstdc++ library.

**Pull request merged**